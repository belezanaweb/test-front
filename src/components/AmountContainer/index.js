import React, { useState, useEffect } from "react";
import axios from 'axios';
import {
    TotalContainer,
    Text, 
} from "./styles";

const TotalAmountContainer = (props) => {
    const [totalAmount, setTotalAmount] =  useState([])    
    
    useEffect(() => {
        const fetchData = async () => {
            const result = await axios.get('http://www.mocky.io/v2/5b15c4923100004a006f3c07'); 
            
            setTotalAmount(result.data)
        }     
        fetchData()
    }, [])

    return(
        <>
            <TotalContainer>
                <Text className="products">PRODUTO</Text>
                <Text className="shipping">FRETE</Text>
                <Text className="productsAmount">R$ {totalAmount.subTotal}0</Text>
                <Text className="shippingAmount">R$ {totalAmount.shippingTotal}0</Text>
                <Text className="discount">DESCONTO</Text>
                <Text className="discountAmount">- R$ {totalAmount.discount},00</Text>
                <Text className="total">TOTAL</Text>
                <Text className="totalAmount">R$ {totalAmount.total}0</Text>                       
            </TotalContainer>                         
        </>
    )    
}
export default TotalAmountContainer