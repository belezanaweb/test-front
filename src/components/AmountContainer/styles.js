import styled from "styled-components";

export const TotalContainer = styled.div`
    height: 100uv;
    width: 341px;
    border: 1px solid #D8D8D8;
    border-radius: 3px;
    margin: 0px 10px 0px 9px;
    display: grid;
    grid-templater-column: 2fr;
    grid-template-row: 4fr;

    @media(min-width: 400px){
        width: 400px;
        background-color: #EEE;
    }
`
export const Text = styled.p`
    height: 17px;
    width: 124px
    color: #212122;
    font-family: "Helvetica Neue, sans-serif";
    font-size: 14px;
    letter-spacing: NaNpx;
    line-height: 17px;
    margin: 0px;
    display: flex;

    @media(min-width: 400px){
        font-size: 16px;
        background-color: #EEE;
    }
                
    &.products {
       grid-collum: 1/2;
       grid-row: 1/2;
       margin: 15px 66px 9px 14px;

    }
    &.shipping {
       grid-collum: 1/2;
       grid-row: 2/3;
       margin: 0px 66px 9px 14px;
    }
    &.productsAmount {
       grid-collum: 2/3;
       grid-row: 1/2;
       margin: 15px 13px 9px 0px;
       justify-content: flex-end;
    }
    &.shippingAmount {
       grid-collum: 2/3;
       grid-row: 2/3;
       margin: 0px 13px 9px 0px;
       justify-content: flex-end;
    }
    &.discount {
        color: #FF7800;
        grid-collum: 1/2;
        grid-row: 3/4;
        margin: 0px 66px 16px 14px;
    }
    &.discountAmount {
        color: #FF7800;
        grid-collum: 2/3;
        grid-row: 3/4;
        margin: 0px 13px 16px 0px;
        justify-content: flex-end;
    }
    &.total {
        font-weight: 700;
        grid-collum: 1/2;
        grid-row: 4/5;
        margin: 0px 66px 13px 14px;
    }
    &.totalAmount {
        font-weight: 700;
        grid-collum: 2/3;
        grid-row: 4/5;
        margin: 0px 13px 13px 0px;
        justify-content: flex-end;
    }
`