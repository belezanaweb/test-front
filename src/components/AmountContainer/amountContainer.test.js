import React from "react";
import { shallow } from "enzyme";
import axios from 'axios';
import AmountContainer from "./index.js";

jest.mock("axios");
const wrapper = shallow(<AmountContainer/>)

describe('AmountContainer', () => {
    it('amount container is rendering correctly', () => {                     
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper).toMatchSnapshot();
    });
    it('should fetch text', async() => {
        const fetchData = () => {
            axios.get('http://www.mocky.io/v2/5b15c4923100004a006f3c07');         
        }
        expect(fetchData).toBeTruthy();
        expect(fetchData).toBeDefined();  
    });
})