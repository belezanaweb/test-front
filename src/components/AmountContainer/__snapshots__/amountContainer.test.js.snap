// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountContainer amount container is rendering correctly 1`] = `ShallowWrapper {}`;
